Engine/Component/Component.h

```diff
@@ -55,8 +56,17 @@ class Component
 	virtual Component* Clone(bool create_on_module = true) const = 0;
 	virtual void Copy(Component * component_to_copy) const = 0;
 
-	virtual void Save(Config& config) const = 0;
-	virtual void Load(const Config &config) = 0;
+	virtual void Save(Config& config) const 
+	{
+		config.AddUInt(UUID, "UUID");
+		config.AddInt((unsigned int)type, "ComponentType");
+		config.AddBool(active, "Active");
+	};
+	virtual void Load(const Config &config)
+	{
+		UUID = config.GetUInt("UUID", 0);
+		active = config.GetBool("Active", true);
+	};
```

Comment:
@mariofv has done this on a different way in the UI branch. He will change it a bit on the merge.